src/tr1/game/lara/cheat.c

```diff
@@ -25,6 +26,39 @@
 
 #include <stdint.h>
 
+static void M_ReinitialiseMeshes(void);
+
+static void M_ReinitialiseMeshes(void)
+{
+    const bool is_gym = Game_GetCurrentLevel()->type == GFL_GYM;
+    for (LARA_MESH mesh = LM_FIRST; mesh < LM_NUMBER_OF; mesh++) {
+        Lara_SwapSingleMesh(
+            mesh, mesh == LM_HEAD || !is_gym ? O_LARA : O_LARA_EXTRA);
+    }
+
+    const LARA_GUN_TYPE back_gun_type =
+        Inv_RequestItem(O_SHOTGUN_ITEM) ? LGT_SHOTGUN : LGT_UNARMED;
+    LARA_GUN_TYPE holsters_gun_type = LGT_UNARMED;
+    if (g_Lara.gun_type != LGT_UNARMED && g_Lara.gun_type != LGT_SHOTGUN) {
+        holsters_gun_type = g_Lara.gun_type;
+    } else if (Inv_RequestItem(O_PISTOL_ITEM)) {
+        holsters_gun_type = LGT_PISTOLS;
+    } else if (Inv_RequestItem(O_MAGNUM_ITEM)) {
+        holsters_gun_type = LGT_MAGNUMS;
+    } else if (Inv_RequestItem(O_UZI_ITEM)) {
+        holsters_gun_type = LGT_UZIS;
+    }
+
+    if (back_gun_type != LGT_UNARMED) {
+        Gun_SetLaraBackMesh(back_gun_type);
+    }
+
+    if (holsters_gun_type != LGT_UNARMED) {
+        Gun_SetLaraHolsterLMesh(holsters_gun_type);
+        Gun_SetLaraHolsterRMesh(holsters_gun_type);
+    }
+}
```

Comment:
This logic is duplicated from somewhere isn't it? I think Lara_InitialiseMeshes. Any chance we can avoid duplication?

Reply:
Yeah, `Lara_InitialiseMeshes` uses `RESUME_INFO` though, so doesn't take into account guns she currently has, plus it results in the original bug with #2461. Maybe we could change the signature though.